h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/DefaultMojoImplTest.java

```diff
@@ -0,0 +1,103 @@
+package hex.tree.xgboost;
+
+import hex.ModelMetricsRegression;
+import hex.SplitFrame;
+import hex.genmodel.MojoModel;
+import hex.genmodel.algos.xgboost.XGBoostJavaMojoModel;
+import hex.genmodel.algos.xgboost.XGBoostMojoReader;
+import hex.genmodel.algos.xgboost.XGBoostNativeMojoModel;
+import hex.genmodel.utils.DistributionFamily;
+import org.junit.*;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runners.Parameterized;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.util.Log;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import hex.tree.xgboost.XGBoostModel.XGBoostParameters.Booster;
+
+public class DefaultMojoImplTest extends TestUtil {
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  @Before
+  public void setupMojoJavaScoring() {
+    System.clearProperty("sys.ai.h2o.xgboost.scoring.java.enable"); // force default behavior
+
+    assertNull(XGBoostMojoReader.getJavaScoringConfig()); // check that MOJO scoring config is not present
+  }
+
+  @Test
+  public void testGBTree() throws IOException {
+    Scope.enter();
+    try {
+      XGBoostModel model = trainModel(Booster.gbtree);
+      MojoModel mojo = model.toMojo();
+      assertEquals(XGBoostJavaMojoModel.class.getName(), mojo.getClass().getName());
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testDART() throws IOException {
+    Scope.enter();
+    try {
+      XGBoostModel model = trainModel(Booster.dart);
+      MojoModel mojo = model.toMojo();
+      assertEquals(XGBoostNativeMojoModel.class.getName(), mojo.getClass().getName());
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  /* FIXME: PUBDEV-6387: gblinear causes a segmenation fault
+  @Test
+  public void testGBLinear() throws IOException {
```

Comment:
Delete it or uncomment it ?

Reply:
This should stay, it cannot be uncommented until PUBDEV-6387 gets fixed.